### PR TITLE
Loop unrolling working on egglog test

### PR DESCRIPTION
**Pull request by oflatt**

### CODE REVIEW

File: tests/snapshots/files__loop_pass_through-optimize.snap [hunk 1]
```diff
@@ -3,7 +3,7 @@ source: tests/files.rs
 expression: visualization.result
 ---
 @main(v0: int) {
-  v1_: int = const 1;
+  v1_: int = const 2;
```
Comment (ezrosent) [inline tests/snapshots/files__loop_pass_through-optimize.snap:6]: Why did this value change? It looks like we're comparing the loop variable against 10, but the rule looks for multiples of 4?

File: dag_in_context/src/optimizations/loop_unroll.egg [hunk 2]
```diff
@@ -23,3 +23,44 @@
           outputs) ;; right now, loop unrolling shares the same outputs, but we could add more context here
         (Arg inputs-ty (NoContext)))))
  :ruleset loop-peel)
+
+
+(function UnrollCtx (Expr Expr) Assumption)
+
+;; unroll a loop with constant bounds and initial value
+(rule
```
Comment (ezrosent) [inline dag_in_context/src/optimizations/loop_unroll.egg:31]: This rule looks reasonable. Question though:

I was imagining that loop unrolling would do something like:

```
do {
   <body>
} while (<pred>)
=>
do {
  <body>
  if (<pred>) <body>
} while(<pred>)
```

I think we'd be able to rewrite bounded-iteration loops this way, and we'd match more often. If we really are comparing (i < C) and (i + 1 < C) ... we'd be able to simplify predicates as well I'd imagine.

Why are we doing this one that hard-codes a multiple of 4? I suspect there's a subtlety I'm missing.

**Review by ezrosent — APPROVED** [review]
couple clarification questions.

File: dag_in_context/src/optimizations/loop_unroll.egg (hunk 2 above)
Comment (oflatt) [reply dag_in_context/src/optimizations/loop_unroll.egg:31]: I'd like to make it work that way instead- I decided to do the simplest thing first

File: tests/snapshots/files__loop_pass_through-optimize.snap (hunk 1 above)
Comment (oflatt) [reply tests/snapshots/files__loop_pass_through-optimize.snap]: Good catch, I'm not sure... I can't reproduce it locally

Comment (oflatt) [reply tests/snapshots/files__loop_pass_through-optimize.snap]: I see now- it's equivalent because you can loop peel the loop once.

Comment (ezrosent) [reply tests/snapshots/files__loop_pass_through-optimize.snap]: But this rule shouldn't do peeling, right? It shouldn't match.

But if you can't repro locally maybe it was a different rule that caused it?

**Pull request merged**